networkx/algorithms/chains.py

```diff
@@ -141,6 +141,11 @@ def _build_chain(G, u, v, visited):
             u, v = v, G.nodes[v]["parent"]
         yield u, v
 
+    # Check if the root is in the graph G. If not, raise a NodeNotFound
+    # error.
```

Comment:
```suggestion
    # Check if the root is in the graph G. If not, raise NodeNotFound
```
very minor change -- only for looks